comp/forwarder/defaultforwarder/internal/retry/telemetry.go

```diff
@@ -308,11 +305,11 @@ func (t onDiskRetryQueueTelemetry) addDeserializeTransactionsCount(count int) {
 }
 
 func toCamelCase(s string) string {
-	caser := cases.Title(language.English)
 	parts := strings.Split(s, "_")
 	var camelCase string
 	for _, p := range parts {
-		camelCase += caser.String(p)
+		camelCase += strings.ToUpper(string(p[0]))
+		camelCase += string(p[1:])
```

Comment:
This will panic if you: 
- call on a string with two underscores. `toCamelCase("thing__thang")`
- call on a string with a trailing underscore, `toCamelCase("thing_thang_")`
- call on a string that starts with an underscore, `toCamelCase("_thing_thang")`

Can we handle these case?

Reply:
The strings are hard coded above in the file, and they all have the expected format, do you still think I should handle those cases ?

Reply:
OK to be fair it’s just about correctly handling the empty string so it’s trivial enough

Reply:
I just handled empty strings, the current code should be equivalent to the previous one, let me know

Reply:
Ah, I didn't fully check the context, but at minimum a comment warning about the edge cases to prevent it being reused somewhere else would be useful.